pkg/ucp/aws/ucpcredentialprovider.go

```diff
@@ -73,19 +86,65 @@ func (c *UCPCredentialProvider) Retrieve(ctx context.Context) (aws.Credentials,
 		return aws.Credentials{}, err
 	}
 
-	if s.AccessKeyCredential == nil || s.AccessKeyCredential.AccessKeyID == "" || s.AccessKeyCredential.SecretAccessKey == "" {
-		return aws.Credentials{}, errors.New("invalid access key info")
-	}
-
-	logger.Info(fmt.Sprintf("Retrieved AWS Credential - AccessKeyID: %s", s.AccessKeyCredential.AccessKeyID))
-
-	value := aws.Credentials{
-		AccessKeyID:     s.AccessKeyCredential.AccessKeyID,
-		SecretAccessKey: s.AccessKeyCredential.SecretAccessKey,
-		Source:          "radiusucp",
-		CanExpire:       true,
-		// Enables AWS SDK to fetch (rotate) access keys by calling Retrieve() after Expires.
-		Expires: time.Now().UTC().Add(c.options.Duration),
+	var value aws.Credentials
+	switch s.Kind {
+	case CredentialKindAccessKey:
+		if s.AccessKeyCredential == nil || s.AccessKeyCredential.AccessKeyID == "" || s.AccessKeyCredential.SecretAccessKey == "" {
```

Comment:
```suggestion
		if s.AccessKeyCredential == nil && (s.AccessKeyCredential.AccessKeyID == "" || s.AccessKeyCredential.SecretAccessKey == "") {
```
I think we will hit nil pointer here when `s.AccessKeyCredential==nil`